src/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.tsx

```diff
@@ -13,38 +13,106 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { Dispatch, useMemo } from 'react';
 import { Progress, WarningPanel } from '@backstage/core-components';
-import { useCortexApi, useEntitiesByTag } from '../../../utils/hooks';
-import { GroupByOption, HeaderType } from '../../../api/types';
+import { useCortexApi } from '../../../utils/hooks';
 import { SingleScorecardHeatmapTable } from './Tables/SingleScorecardHeatmapTable';
+import {
+  StringIndexable,
+  applyScoreFilters,
+  catalogToRelationsByEntityId,
+} from './HeatmapUtils';
+import { HomepageEntity } from '../../../api/userInsightTypes';
+import { HeatmapPageFilters, SortBy } from './HeatmapFilters';
+import { ScorecardLadder } from '../../../api/types';
+import { keyBy } from 'lodash';
 
 interface SingleScorecardHeatmapProps {
   entityCategory: string;
+  entitiesByTag: StringIndexable<HomepageEntity>;
   scorecardId: number;
-  groupBy: GroupByOption;
-  headerType: HeaderType;
+  ladder: ScorecardLadder | undefined;
+  filters: HeatmapPageFilters;
+  setFiltersAndNavigate: Dispatch<React.SetStateAction<HeatmapPageFilters>>;
+  sortBy?: SortBy;
+  setSortBy: Dispatch<React.SetStateAction<SortBy | undefined>>;
+  tableHeight: number;
 }
 
 export const SingleScorecardHeatmap = ({
   entityCategory,
+  entitiesByTag,
   scorecardId,
-  groupBy,
-  headerType,
+  ladder,
+  setFiltersAndNavigate,
+  filters,
+  sortBy,
+  setSortBy,
+  tableHeight,
 }: SingleScorecardHeatmapProps) => {
+  const {
+    groupBy,
+    headerType,
+    scoreFilters,
+    useHierarchy,
+    hideWithoutChildren,
+  } = filters;
   const {
     value: scores,
     loading: loadingScores,
     error: scoresError,
   } = useCortexApi(api => api.getScorecardScores(scorecardId), [scorecardId]);
 
-  const { value: ladders, loading: loadingLadders } = useCortexApi(
-    api => api.getScorecardLadders(scorecardId),
-    [scorecardId],
-  );
-  const { entitiesByTag, loading: loadingEntities } = useEntitiesByTag();
+  const { ownerEmailByEntityId, groupTagByEntityId } = useMemo(() => {
+    return catalogToRelationsByEntityId(entitiesByTag);
+  }, [entitiesByTag]);
+
+  const {
+    value: domainIdByEntityId,
+    error: domainByEntityError,
+    loading: isLoadingDomainByEntity,
+  } = useCortexApi(api => api.getEntityDomainAncestors());
+  const domainTagByEntityId = useMemo(() => {
+    if (!domainIdByEntityId) {
+      return {};
+    }
+
+    const map = {} as Record<string, string[]>;
+    const domainsById = keyBy(
+      Object.values(entitiesByTag).filter(entity => entity.type === 'domain'),
+      domain => domain.id,
+    );
+
+    Object.keys(domainIdByEntityId.entitiesToAncestors).forEach(entityId => {
+      map[entityId] = domainIdByEntityId.entitiesToAncestors[
+        parseInt(entityId, 10)
+      ].map(parentId => domainsById[parentId].codeTag);
```

Comment:
this breaks our integration, it fails with: 
```
TypeError
Cannot read properties of undefined (reading 'codeTag')
Call Stack
 eval
  node_modules/@cortexapps/backstage-plugin/dist/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.esm.js:62:126
 eval
  node_modules/@cortexapps/backstage-plugin/dist/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.esm.js:62:86
 eval
  node_modules/@cortexapps/backstage-plugin/dist/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.esm.js:61:57
 updateMemo
  node_modules/react-dom/cjs/react-dom.development.js:16427:19
 Object.useMemo
  node_modules/react-dom/cjs/react-dom.development.js:17067:16
 useMemo
  node_modules/react/cjs/react.development.js:1651:21
 SingleScorecardHeatmap
  node_modules/@cortexapps/backstage-plugin/dist/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.esm.js:52:77
 renderWithHooks
  node_modules/react-dom/cjs/react-dom.development.js:15486:18
◀ Prev× CloseNext ▶
```
<img width="988" alt="Screenshot 2024-06-24 at 10 47 26" src="https://github.com/cortexapps/backstage-plugin/assets/346576/67ac5dbf-335d-498c-a7a3-061b4e9ffd45">